File: backend/aci/common/schemas/quota.py
from pydantic import BaseModel

from aci.common.schemas.plans import PlanFeatures


class QuotaResourceUsage(BaseModel):
    """single resource quota usage"""

    used: int
    limit: int
    remaining: int


class PlanInfo(BaseModel):
    """plan info"""

    name: str
    features: PlanFeatures


class QuotaUsageResponse(BaseModel):
    """complete quota usage response"""

    projects: QuotaResourceUsage
    linked_accounts: QuotaResourceUsage  # unique account owner IDs across org
    agents_credentials: QuotaResourceUsage  #  total org secrets in secrets table
    plan: PlanInfo

File: frontend/src/lib/types/quota.ts
export interface QuotaUsage {
  projects: {
    used: number;
    limit: number;
  };

  linked_accounts: {
    used: number;
    limit: number;
  };

  agent_credentials: {
    used: number;
    limit: number;
  };

  plan: {
    name: string;
    features: {
      projects: number;
      linked_accounts: number;
      api_calls_monthly: number;
      agent_credentials: number;
      developer_seats: number;
      custom_oauth: boolean;
      log_retention_days: number;
    };
  };
}
